prov/gni/src/gnix_cq.c

```diff
@@ -327,26 +340,25 @@ static int verify_cq_attr(struct fi_cq_attr *attr)
 static int gnix_cq_close(fid_t fid)
 {
 	struct gnix_fid_cq *cq;
-	struct slist_entry *entry;
-	struct gnix_cq_entry *item;
 
 	cq = container_of(fid, struct gnix_fid_cq, cq_fid);
-	if (atomic_get(&cq->ref_cnt) != 0)
+	if (atomic_get(&cq->ref_cnt) != 0) {
+		GNIX_INFO(FI_LOG_CQ, "CQ ref count: %d, not closing.\n",
+			  cq->ref_cnt);
 		return -FI_EBUSY;
+	}
 
 	atomic_dec(&cq->domain->ref_cnt);
-	assert(atomic_get(&cq->domain->ref_cnt) > 0);
+	assert(atomic_get(&cq->domain->ref_cnt) >= 0);
 
```

Comment:
While this doesn't lead to incorrect behavior, when assertions are enabled, it's possible that more than one thread hits the assertion, which might make debugging a little more difficult.